### PR TITLE
G5/feature/material card

### PR DESCRIPTION
**Pull request by RuKaSuu**
Add the modal for the edit + borrowing history.

### CODE REVIEW

**Review by AyakorK — APPROVED** [review]
LGTM @Maghwyn I let you watch those files to see if it also looks good to you

File: back-end/src/base/materials/materials.controller.ts [hunk 1]
```diff
@@ -35,11 +35,11 @@ export class MaterialsController {
 		});
 	}
 	@Put('/update/:id')
-	updateMaterial(@Req() req, @Res() res: Response) {
+	updateMaterial(@Req() req) {
```
Comment (Maghwyn) [inline back-end/src/base/materials/materials.controller.ts]: @Res() res: Response is mandatory.

File: back-end/src/base/materials/materials.controller.ts [hunk 2]
```diff
@@ -69,9 +69,27 @@ export class MaterialsController {
...
 	}
 
-	@Get('/:id')
-	async getMaterialById(@Param('id') id: string, @Res() res: Response) {
-		const material = await this.materialsService.getMaterialById(id);
-		res.status(200).json(material);
+	@Get('get/:id')
+	getMaterialById(@Req() req, @Res() res: Response) {
+		const id = req.params.id;
+		this.materialsService.getMaterialById(id).then((response) => {
```
Comment (Maghwyn) [inline back-end/src/base/materials/materials.controller.ts]: Ne pas utiliser de .then

Faîtes plutôt await service.
return res.status...

File: front-end/src/components/materials/DetailsMaterials.vue [hunk 3]
```diff
@@ -1,20 +1,267 @@
...
+					console.log(err);
+				});
+		};
+
+		const deleteMaterial = () => {
+			http
+				.delete(`materials/delete/` + props.id)
+				.then((res) => {
+					//Reload the page
+					window.location.reload();
```
Comment (Maghwyn) [inline front-end/src/components/materials/DetailsMaterials.vue]: Pas de reload, on est en vue.js

File: front-end/src/components/materials/DetailsMaterials.vue [hunk 4]
```diff
@@ -1,20 +1,267 @@
...
+		};
+
+		const editMaterial = () => {
+			http
+				.put(`materials/update/${props.id}`, {
+					...material.value,
+				})
+				.then((res) => {
+					//Reload the page
+					window.location.reload();
```
Comment (Maghwyn) [inline front-end/src/components/materials/DetailsMaterials.vue]: Same

**Review by Maghwyn — CHANGES_REQUESTED** [review]

**Pull request merged**